### ISSUE TITLE
`mx.np.outer` does not support integer inputs

### ISSUE THREAD

**Issue by haojin2** [apache/incubator-mxnet#17083]
## Error
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/incubator-mxnet/python/mxnet/numpy/multiarray.py", line 918, in __repr__
    array_str = self.asnumpy().__repr__()
  File "/home/ubuntu/incubator-mxnet/python/mxnet/ndarray/ndarray.py", line 2552, in asnumpy
    ctypes.c_size_t(data.size)))
  File "/home/ubuntu/incubator-mxnet/python/mxnet/base.py", line 278, in check_call
    raise MXNetError(py_str(_LIB.MXGetLastError()))
mxnet.base.MXNetError: [06:26:13] ../src/operator/numpy/./np_tensordot_op-inl.h:605: Check failed: out.type_flag_ == kFloat32 || out.type_flag_ == kFloat64 || (out.type_flag_ == kFloat16 && ctx.run_ctx.ctx.dev_mask() == mshadow::gpu::kDevMask): Tensordot only supports float32/float64 for CPU, and float16/float32/float64 for GPU
```
## Reproduction
```python
from mxnet import np,npx
npx.set_np()

a = np.random.uniform(-128, 128, size=(100,)).astype(np.int64)
b = np.outer(a, a)
```

**Comment by yzhliu** [comment]
(Tentatively) @AntiZpvoh will be helping once he has finished ongoing tasks.

**Comment by Yiyan66** [comment]
fixed by #17510
